lib/pbench/server/api/resources/query_apis/datasets_detail.py

```diff
@@ -158,12 +157,11 @@ def postprocess(self, es_json: JSON, context: CONTEXT) -> JSON:
         try:
             m = self._get_metadata(src["run"]["name"], context["metadata"])
         except DatasetNotFound:
-            abort(
-                HTTPStatus.BAD_REQUEST,
-                message=f"Dataset {src['run']['name']} not found",
+            raise APIAbort(
+                HTTPStatus.NOT_FOUND, f"Dataset {src['run']['name']} not found"
             )
         except MetadataError as e:
-            abort(HTTPStatus.BAD_REQUEST, message=str(e))
+            raise APIAbort(HTTPStatus.BAD_REQUEST, str(e))
```

Comment:
Why did we change from `BAD_REQUEST` to `NOT_FOUND` just for this one case?  (The other failure cases in this function return `BAD_REQUEST`....)  And, it's not that the _dataset_ is not found, right?  The issue is that the server-side metadata for this dataset is not found...isn't that weirder than `NOT_FOUND`?

Reply:
You're right, and maybe this really should be an `HTTPStatus.INTERNAL_SERVER_ERROR` because I don't think it implies any problem at all in the client's inputs.

There are several more `BAD_REQUEST` errors up above (138, 143) in the same category. We presumably already validated the dataset name, made the Elasticsearch request, and we're complaining about the state or lack of Elastic data by blaming it on the client when that's clearly an internal problem.

So I'll agree that the change in this PR from `BAD_REQUEST` to `NOT_FOUND` is probably not what we want; however reverting that change isn't the answer, and I think the correct fix is that most of these should be changed to `INTERNAL_SERVER_ERROR`. I think that ought to be handled as a separate bug Jira rather than stretch out this PR.

Reply:
OK, let's not (unduly) stretch out this PR.  But, are we happy with changing this one `BAD_REQUEST` to `NOT_FOUND` or should it be reverted for this PR?